sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -239,13 +263,45 @@ private void processTextEventStream(HttpRequest httpRequest, HttpURLConnection c
             }
         } catch (IOException e) {
             throw LOGGER.logThrowableAsError(new UncheckedIOException(e));
+        } finally {
+            connection.disconnect();
         }
     }
 
     private static boolean isTextEventStream(Headers responseHeaders) {
         return Objects.equals(ContentType.TEXT_EVENT_STREAM, responseHeaders.getValue(HeaderName.CONTENT_TYPE));
     }
 
+    private void setResponseBodySupplier(HttpResponse<?> httpResponse, HttpURLConnection connection) {
+        HttpResponseAccessHelper.setBodySupplier(httpResponse, () -> {
+            try {
+                return BinaryData.fromByteBuffer(getAccessibleByteArrayOutputStream(connection).toByteBuffer());
```

Comment:
I'm not sure this is technically streaming, we're just deferring the eager reading of the body until the supplier is called.

This could also break in interesting ways if `HttpResponse.getBody` is called from multiple threads or before the `body` variable gets updated in memory if one thread sets it and then another thread tries to access it.

Reply:
You're right, I'll change it to
```java
return BinaryData.fromStream(connection.getInputStream());
```